tests/test_interventions.py

```diff
@@ -1203,7 +1236,148 @@ def test_quarantined_have_trace_token(self, test_params, time_steps_test ):
+        params = utils.set_app_users_fraction_all( params, app_users_fraction)
+        params = utils.set_priority_test_contacts_all( params, priority_test_contacts )  
+        model  = utils.get_model_swig( params )
+
+        print(f"test_time {test_time}")
+        print(f"end_time {params.get_param('end_time')}")
+        print(f"test_order_wait {params.get_param('test_order_wait')}")
+        print(f"test_result_wait {params.get_param('test_result_wait')}")
+        print(f"test_order_wait_priority {params.get_param('test_order_wait_priority')}")
+        print(f"test_result_wait_priority {params.get_param('test_result_wait_priority')}")
+        
+        # step through time
+        for time in range( test_time ):
+          model.one_time_step()
+          
+        # write files
+        model.write_trace_tokens()
+        model.write_individual_file()
+        model.write_transmissions()
+
+        # read CSV's
+        pre_df_trace = pd.read_csv( constant.TEST_TRACE_FILE, comment="#", sep=",", skipinitialspace=True )
+        pre_df_trans = pd.read_csv( constant.TEST_TRANSMISSION_FILE, sep = ",", comment = "#", skipinitialspace = True )
+        pre_df_indiv = pd.read_csv( constant.TEST_INDIVIDUAL_FILE, comment="#", sep=",", skipinitialspace=True )
+        
+        # determine who should get a priority test:
+        # 1. get the people who developed symptoms on the last step and have the app
+        pre_df_symp = pre_df_trans[(pre_df_trans["time_symptomatic"] == test_time)]
+        pre_df_symp = pd.merge( pre_df_symp, pre_df_indiv, left_on = "ID_recipient", right_on = "ID", how = "left")
+        pre_df_symp = pre_df_symp[ (pre_df_symp["app_user"] == True ) ]
+        pre_df_symp = pre_df_symp.loc[:,["ID_recipient","test_status"]]
+ 
+        # 2. get the number of contacts using the trace_tokens
+        pre_df_trace = pre_df_trace[ ( pre_df_trace["index_time"] == test_time) ]
+        pre_df_trace_sum = pre_df_trace.groupby("index_ID").size().reset_index(name="n_interactions")
+        pre_df_symp  = pd.merge( pre_df_symp, pre_df_trace_sum, left_on = "ID_recipient", right_on = "index_ID", how = "left" )
+        pre_df_symp["priority"] = ( pre_df_symp["n_interactions"] > priority_test_contacts )
+
+        # check we have sufficient priority and non-priority to test
+        if ( ( params.get_param( "test_order_wait_priority" ) != -1 ) and ( params.get_param( "test_result_wait_priority" ) != -1 ) ):
+            np.testing.assert_equal(sum(pre_df_symp["priority"]== True) > 30, True, "Not sufficient priority cases to meaningfully test" )
+        np.testing.assert_equal(sum(pre_df_symp["priority"]== False) > 30, True, "Not sufficient non-priority cases to meaningfully test" )
+        
+        # check the test results being received are in line with whether it is a priority test
+        if ( params.get_param( "test_order_wait_priority" ) != -1 and params.get_param( "test_result_wait_priority" ) != -1 ):
```

Comment:
This currently only works if `test_order_wait_priority == 0`. So, you need to look at people who become symptomatic `test_order_wait_priority` steps prior to `test_time` at line 1305. I expect it is a one line change. One of your tests parameter sets should have `test_order_wait=2` and `test_order_wait_priority = 1`.

Reply:
@roberthinch made this change and now I'm getting a seg fault, could you please take a look at the test? I uploaded the latest version